### ISSUE TITLE
setting WindowMode.maximized does not create a maximized window

### ISSUE THREAD

**Issue by jDomantas** [ggez/ggez#925]
Setting `.maximized(true)` for WindowMode creates a window that is 800x600 px instead of taking the full screen.

**To Reproduce**

```rust
struct MainState;

impl ggez::event::EventHandler for MainState {
    fn update(&mut self, _ctx: &mut ggez::Context) -> ggez::GameResult {
        Ok(())
    }

    fn draw(&mut self, ctx: &mut ggez::Context) -> ggez::GameResult {
        ggez::graphics::clear(ctx, ggez::graphics::Color::from_rgb(50, 100, 200));
        ggez::graphics::present(ctx)?;
        Ok(())
    }
}

pub fn main() -> ggez::GameResult {
    let cb = ggez::ContextBuilder::new("maximized", "ggez")
        .window_mode(ggez::conf::WindowMode::default()
            .resizable(true)
            .maximized(true));
    let (ctx, event_loop) = &mut cb.build()?;

    let state = &mut MainState;
    ggez::event::run(ctx, event_loop, state)
}
```

**Expected behavior**
Created window is maximized and takes up the whole screen.

**Actual behavior**

![image](https://user-images.githubusercontent.com/8329015/115142828-7e308600-a04c-11eb-9337-bd9e712f4cc4.png)

Note that the resize button is the kind that is on maximized windows - if I click it the window goes back to "normal" mode (initial size, somewhere in the middle-ish of the screen) and clicking it again makes the window actually maximized.

The workaround I found is to initially set the window size to some very large value. Then it eventually ends up being the needed size to be maximized, but it does so by going through multiple resize steps. In my case I get resize events of ~60% of the screen, then slightly larger than the screen, and then only on the third step it ends up being the needed size.

**Hardware and Software:**
 - ggez version: 0.5.1
 - OS: Windows 10
 - Graphics card: NVidia GeForce GTX 1060
 - Graphics card drivers: no idea

**Comment by PSteinhaus** [comment]
Hmm, this is not good.
winit has been updated in the current devel branch (which doesn't build right now, but #917 builds), so it'd be interesting to check if the behavior changes there.

If it's still broken then we probably need to rewrite ```graphics::context::GraphicsContextGeneric::set_window_mode``` to actively ignore window dimensions (i.e. setting them to be unlimited) when the maximized flag is set.

What SHOULD work is using the fullscreen options instead, but i of course don't know whether this would work for you, because it doesn't give you a maximized window, but a fullscreen window of course.

**Comment by jDomantas** [comment]
Tried out with that branch, and it still does not work (but differently). If I set the window mode to maximized immediately then the window blinks to maximized and then immediately back to normal mode (in the middle of the screen, the button is like on a non-maximized window). It also gives 4 resize events: first one is for maximized size, second is requested size, third is maximized again, and the fourth is requested size. If I set it to maximized sometime later in the `update` then title bar icon blinks for a frame but otherwise the window does not move or change. This one also yields 4 resize events, but second one is for initial size, and fourth one is for new requested size. Maximizing by just clicking the button works as expected.

Desktop fullscreen in 0.5.1 also seems to be broken - I get one of two results (not even changing anything, just rerunning the same binary again):

* I get a window that looks fullscreen, but according to resize event is 1936x1100 (my screen is 1920x1080)
* I get a window that is not placed correctly, has wrong resolution, and is also below the task bar (the terminal window is drawn on top of everything, so that bit is expected):

![image](https://user-images.githubusercontent.com/8329015/115151782-180d2880-a077-11eb-9219-3d57be14b236.png)

Desktop fullscreen mode does seem to work and resize the window correctly when I tried it with the pull request branch.

**Comment by PSteinhaus** [comment]
Yes, fullscreen is a bit broken and was (hopefully) fixed in 0.6.

Some of that resizing probably happens because of the hidpi-factor of your system. On Windows 10 it can be set in the display settings (under "Scale and layout"). Setting it back to 100% will probably reduce the amount of resizing events (at least in 0.5.1) but not solve the problem...

I was able to reproduce the behavior you describe. I think I know where the 4 events may come from:
Inside of ```set_window_mode``` we call ```window.set_inner_size``` after calling ```window.set_maximized(mode.maximized)```, which is probably a mistake, because ```set_inner_size``` is said to unmaximize maximized windows. This triggers the second sizing event. The whole thing happens once in the creation of the window and then when the event loop starts working (that part is only a guess), thereby creating 2x2=4 resize events.

And you know what? I was able to reduce that to just 1 event by moving ```window.set_maximized(mode.maximized)``` from the start of ```set_window_mode``` to here:
```rust
match mode.fullscreen_type {
            FullscreenType::Windowed => {
                window.set_fullscreen(None);
                window.set_decorations(!mode.borderless);
                window.set_inner_size(dpi::PhysicalSize {
                    width: f64::from(mode.width),
                    height: f64::from(mode.height),
                });
                window.set_resizable(mode.resizable);
                window.set_maximized(mode.maximized);
            }
            FullscreenType::True => {
            ...
```

😄 
Thanks for pointing out this bug. The occurance of too many resize events has been something that annoyed me for some time, so it's nice to be able to finally improve this part a little :)
